invoke-ai/InvokeAI
# feat: updateable workflow nodes

## What type of PR is this? (check all applicable)



## Have you discussed this change with the InvokeAI team?

      
## Have you updated all relevant documentation?


## Description

[fix(nodes): bump version of nodes post-pydantic v2](https://github.com/invoke-ai/InvokeAI/commit/5cb3fdb64ccd9464e61221d54d163b148937b7bd)

This was not done, despite new metadata fields being added to many nodes.

[feat(ui): add update node functionality](https://github.com/invoke-ai/InvokeAI/commit/3f6e8e9d6b9b9303cf3647a7bae65ee2b8880037)

A workflow's nodes may update itself, if its major version matches the template's major version.

If the major versions do not match, the user will need to delete and re-add the node (current behaviour).

The update functionality is not automatic (for now). The logic to update the node is pretty simple, but I want to ensure it works well first before doing it automatically when a workflow is loaded.

- New `Details` tab on Workflow Inspector, displays node title, type, version, and notes
- Button to update the node is displayed on the `Details` tab
- Add hook to determine if a node needs an update, may be updated (i.e. major versions match), and the callback to update the node in state
- Remove the notes modal from the little info icon
- Modularize the node building logic

## Related Tickets & Documents



Probably exist but not sure where.

## QA Instructions, Screenshots, Recordings

Load an old workflow with nodes that need to be updated. Click on each node that needs updating and click the update button. Workflow should work.
